pilot/pkg/model/service.go

```diff
@@ -401,6 +401,10 @@ type ServiceAttributes struct {
 	// Used by the aggregator to aggregate the Attributes.ClusterExternalAddresses
 	// for clusters where the service resides
 	ClusterExternalAddresses map[string][]string
+
+	// The ID for multi clusters.
+	// Currently only used by service event handler.
+	ClusterID string
```

Comment:
This seems like it will lead to bugs... we just set it right before we call the service event handler? and then its set forever? Its very hard to understand when this is set and when it is not

Reply:
Yeah, so to prevent missing setting, i set this value in aggregator https://github.com/istio/istio/pull/18882/files#diff-09a72f0330de5bdfca01320b83c4817eR300.
I am not very sure if this is the best place.

Reply:
@howardjohn  I want to know your idea.